waspc/src/Wasp/Analyzer/Parser/Token.hs

```diff
@@ -26,6 +15,31 @@ data TokenType
   | TEOF
   deriving (Eq, Show)
 
+data SpecialChar
```

Comment:
I think TypeScript uses the word `Punctuator` for these guys. Other sources split them into two groups of special strings `Operators` and `Separators`.

> So maybe we could just call them all symbols. Although it is useful to have this distrniction between symbol and keyword for error reporting since keywords are usually also valid identifiers while symbols are often not.

This comment digs into what I don't like about this solution. The "Project name cannot be a keyword" error (and the fact that keywords can lexically be identifiers) isn't obvious from the code. It's something we (as humans) noticed and added to the error message, so the solution still doesn't seem quite right. I'm having a difficult time explaining it, but I hope you get it.

Reply:
Operators vs Separators -> interesting, and it makes sense. I do like Symbols though, as a unifying name, but probably it is best to stick with how others do it.

I get what you mean with the keywords. It is an assumption that we don't enforce in any way. But it is so common - the meaning of "keyword" has "word" in it which means we humans expect it to be a word, which well then be valid identifier - that it is relatively safe to expect it will not be violated.